src/app/pages/TransactionDetailPage/index.tsx

```diff
@@ -341,6 +372,54 @@ export const TransactionDetailView: FC<{
 
           <dt>{t('common.gasLimit')}</dt>
           <dd>{transaction.gas_limit.toLocaleString()}</dd>
+
+          {transaction.encryption_envelope && (
+            <>
+              <dt>{t('transactions.encryption.format')}</dt>
+              <dd>
+                {transaction.encryption_envelope.format} &nbsp; <TransactionEncrypted />
+              </dd>
+
+              <dt>{t('transactions.encryption.publicKey')}</dt>
+              <dd>
+                <Typography variant={'mono'} sx={{ overflowWrap: 'anywhere' }}>
+                  {transaction.encryption_envelope.public_key}
+                </Typography>
+              </dd>
+
+              <dt>{t('transactions.encryption.dataNonce')}</dt>
+              <dd>
+                <Typography variant={'mono'} sx={{ overflowWrap: 'anywhere' }}>
+                  {transaction.encryption_envelope.data_nonce}
+                </Typography>
+              </dd>
+
+              {transaction.encryption_envelope.data !== undefined && (
```

Comment:
Why we need to check `undefined` strictly, isn't `transaction.encryption_envelope.data && ` enough? Why we don't have similar checks for other optional string props like data_nonce, result_nonce etc ?

Reply:
I am testing for `undefined` because in seme cases, an empty string might be a valid result, and I don't want to hide it. You are right about the missing tests for other fields, adding them...